test-framework/common/src/main/java/io/quarkus/test/common/TestResourceManager.java

```diff
@@ -82,6 +83,23 @@ public void close() {
         }
     }
 
+    public void cleanup(Object testInstance) {
+        for (QuarkusTestResourceLifecycleManager testResource : testResources) {
+            if (!QuarkusTestResourceWithCleanupLifecycleManager.class.isAssignableFrom(testResource.getClass())) {
```

Comment:
Why is this needed?

Reply:
I am checking if the instance of QuarkusTestResourceLifecycleManager is actually also the QuarkusTestResourceWithCleanupLifecycleManager subclass - if I can count on the cleanup method being there. If it's not, there's nothing to do.

Reply:
ah, should have been continue, right :-)

Reply:
Oh, sorry, I didn't see that you introduced a new type.

Reply:
But that brings up the question, why introduce a new type at all? Why not just add a default `cleanup` method to the existing `QuarkusTestResourceLifecycleManager`?

Reply:
It is a possibility, yes.
And one more thing I see here wrong - line 96: 
```
.anyMatch(it -> QuarkusTestResourceWithCleanupLifecycleManager.class.isAssignableFrom(it.value())))
```
should rather be
```
.anyMatch(it -> testResourceWithCleanup.getClass().isAssignableFrom(it.value())))
```
to clean just the resources that have been used for the given testInstance (based on the annotations).

Reply:
I really don't see any value in providing an extra `QuarkusTestResourceWithCleanupLifecycleManager`. Unless I am missing something, it just complicated things.